src/main/java/lotto/dto/LottoResultAggregation.java

```diff
@@ -0,0 +1,35 @@
+package lotto.dto;
+
+import static java.util.stream.Collectors.groupingBy;
+
+import java.util.List;
+import java.util.Map;
+
+import lotto.LottoMatchResult;
+import lotto.model.LottoReward;
+
+public class LottoResultAggregation {
+    private final Map<LottoReward, List<LottoMatchResult>> rewardAndCountMaps;
+
+    public LottoResultAggregation(List<LottoMatchResult> lottoMatchResults) {
+        this.rewardAndCountMaps = lottoMatchResults.stream()
+                                                   .filter(LottoMatchResult::rewardable)
+                                                   .collect(groupingBy(lottoMatchResult -> lottoMatchResult.getLottoReward().get()));
+    }
+
+//    public LottoResultAggregation(LottoMatchResults lottoMatchResults) {
+//        this.rewardAndCountMaps = lottoMatchResults.aggregate();
+//    }
+
+    public LottoResultAggregation(Map<LottoReward, List<LottoMatchResult>> rewardAndCountMaps) {
+        this.rewardAndCountMaps = rewardAndCountMaps;
+    }
+
+    public int rewardMatchCount(LottoReward lottoReward) {
+        if (rewardAndCountMaps.containsKey(lottoReward)) {
+            return rewardAndCountMaps.get(lottoReward).size();
+        }
+
+        return 0;
```

Comment:
숫자 리터럴 `0`과 같은 매직 넘버는 상수로 선언하여 의미를 부여해보면 어떨까요?
아니면 `Map`의 `getOrDefault()`를 활용하는 것도 하나의 방법이 될 것 같습니다.
```suggestion
        return rewardAndCountMaps.getOrDefault(lottoReward, Collections.emptyList()).size();
```